### PR TITLE
Add tests for annotation types

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Updates config.
  * Removes redundant closing tags for `<li>` used in Javadocs.
  * Adds tests for otherwise unused annotation types.

### CODE REVIEW

File: src/test/kotlin/io/spine/annotation/AnnotationsTest.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import io.kotest.matchers.shouldBe
+import java.lang.annotation.RetentionPolicy.SOURCE
+import org.junit.jupiter.api.Test
+
+/**
+ * The mere purpose of this test suite is making sure that
+ * the annotations (which are not used in the production code) are present in tests.
+ */
+internal class AnnotationsTest {
```
Comment (yevhenii-nadtochii) [inline src/test/kotlin/io/spine/annotation/AnnotationsTest.kt:37]: It seems to lack `@DisplayName`.

**Review by yevhenii-nadtochii — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/io/spine/internal/dependency/Kotest.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ package io.spine.internal.dependency
  * @see <a href="https://kotest.io/">Kotest site</a>
  */
 object Kotest {
-    const val version = "5.6.2"
+    const val version = "5.6.1"
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/internal/dependency/Kotest.kt:37]: Let's review this version downgrade. I am not sure it's the direction we want.

File: buildSrc/src/main/kotlin/io/spine/internal/dependency/Spine.kt [hunk 3]
```diff
@@ -40,62 +40,98 @@ object Spine {
...
         const val reflect = "2.0.0-SNAPSHOT.182"
 
-        /** The version of [Spine.logging]. */
-        const val logging = "2.0.0-SNAPSHOT.191"
+        /**
+         * The version of [Spine.logging].
+         *
+         * @see <a href="https://github.com/SpineEventEngine/logging">spine-logging</a>
+         */
+        const val logging = "2.0.0-SNAPSHOT.186"
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/internal/dependency/Spine.kt:62]: Again, this is a downgrade. Please review.

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov please see my comments.

File: buildSrc/src/main/kotlin/io/spine/internal/dependency/Spine.kt (hunk 3 above)
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/io/spine/internal/dependency/Spine.kt]: @yevhenii-nadtochii, the most recent version which can be used here is `2.0.0-SNAPSHOT.191`. After that the build complains about the lack of `flogger:api` dependency:

```
* What went wrong:
Could not determine the dependencies of task ':extractIncludeProto'.
> Could not resolve all files for configuration ':compileProtoPath'.
   > Could not find flogger:api:.
     Required by:
         project : > io.spine:spine-logging:2.0.0-SNAPSHOT.192 > io.spine:spine-logging-jvm:2.0.0-SNAPSHOT.192
```

**Review by armiol — APPROVED** [review]

**Review by yevhenii-nadtochii — APPROVED** [review]

**Pull request merged**